### PR TITLE
Posts: fix posts warnings

### PR DESCRIPTION
**Pull request by rralian**
While checking out an unrelated issue, I noticed quite a few warnings in the console when visiting the posts page for all-sites, jetpack sites, and wpcom sites.

posts warnings

## testing
Just go to http://calypso.localhost:3000/posts with your console open and make sure you're not seeing any of the warnings from the screenshot above.

### CODE REVIEW

**Comment by retrofox** [comment]
Taking a look.

File: client/my-sites/posts/post.jsx [hunk 1]
```diff
@@ -278,7 +278,7 @@ module.exports = React.createClass({
 
 		if ( showLikes ) {
 			if ( post.like_count > 0 ) {
-				likeTitle = this.translate( '1 Like', '%(count)s Likes', {
+				likeTitle = this.translate( '%(count)s Like', '%(count)s Likes', {
```
Comment (jordwest) [inline client/my-sites/posts/post.jsx]: Is this change related to the warnings?

**Comment by jordwest** [comment]
:+1: The code changes look good, also confirmed nice clean console :smiley: 
screen shot 2016-01-14 at 5 21 33 pm

File: client/my-sites/posts/post.jsx (hunk 1 above)
Comment (retrofox) [inline client/my-sites/posts/post.jsx]: No, this is not @jordwest.
The changes that remove those warnings are made in `posts-navigation.jsx` file.

File: client/my-sites/posts/posts-navigation.jsx [hunk 2]
```diff
@@ -369,14 +369,10 @@ export default React.createClass( {
...
 	 */
 	getCountByStatus( status ) {
-		var count = false;
-		if ( false !== this.state.counts[ status ] ) {
-			count = this.numberFormat( this.state.counts[ status ] );
-		}
-
-		return count;
+		let count = this.state.counts[ status ];
+		return ( count !== false ) ? count : null;
```
Comment (retrofox) [inline client/my-sites/posts/posts-navigation.jsx]: A simple note here, are these parentheses is needed here ?

**Comment by retrofox** [comment]
LGFM

File: client/my-sites/posts/post.jsx (hunk 1 above)
Comment (rralian) [inline client/my-sites/posts/post.jsx]: Actually, this code was causing a warning, but not in the browser console... but rather in the terminal via i18nlint.

eslint warning

File: client/my-sites/posts/posts-navigation.jsx (hunk 2 above)
Comment (rralian) [inline client/my-sites/posts/posts-navigation.jsx]: No, but I like 'em. I prefer to use parens in ternaries... it helps me to more immediately make sense of the line at a glance.
